delfin/drivers/netapp/dataontap/netapp_handler.py

```diff
@@ -115,39 +124,45 @@ def get_storage(self):
                 constant.STORAGE_STATUS_COMMAND)
             controller_info = self.ssh_do_exec(
                 constant.CONTROLLER_SHOW_DETAIL_COMMAND)
-            controller_array = controller_info.split(
-                constant.CONTROLLER_SPLIT_STR)
-            Tools.split_value_map(controller_array[1], controller_map, ":")
-            version_array = version_info.split('\r\n')
-            version = version_array[0].split(":")
-            status = constant.STORAGE_STATUS.get(
-                status_info.split("\r\n")[2])
+            Tools.split_value_map_list(
+                controller_info, controller_map_list, ":")
+            version_array = version_info.split("\r\n")
+            storage_version = ''
+            for version in version_array:
+                if 'NetApp' in version:
+                    storage_version = version.split(":")
+                    break
+            status = self.get_table_data(status_info)
+            status = constant.STORAGE_STATUS.get(status[0].split()[0])
             disk_list = self.get_disks(None)
             pool_list = self.list_storage_pools(None)
-            storage_map = {}
-            Tools.split_value_map(system_info, storage_map, split=':')
-            for disk in disk_list:
-                raw_capacity += disk['capacity']
-
-            for pool in pool_list:
-                total_capacity += pool['total_capacity']
-                free_capacity += pool['free_capacity']
-                used_capacity += pool['used_capacity']
-
-            storage_model = {
-                "name": storage_map['ClusterName'],
-                "vendor": constant.STORAGE_VENDOR,
-                "model": controller_map['Model'],
-                "status": status,
-                "serial_number": storage_map['ClusterSerialNumber'],
-                "firmware_version": version[0],
-                "location": controller_map['Location'],
-                "total_capacity": total_capacity,
-                "raw_capacity": raw_capacity,
-                "used_capacity": used_capacity,
-                "free_capacity": free_capacity
-            }
-            return storage_model
+            storage_map_list = []
+            Tools.split_value_map_list(
+                system_info, storage_map_list, split=':')
+            if len(storage_map_list) > 0:
+                storage_map = storage_map_list[len(storage_map_list) - 1]
```

Comment:
For line 143 & 145, if you want to get the last one of a list, you can do it by `storage_map = storage_map_list[-1]`

Reply:
The modified